### DISCUSSION TITLE
RU Auth-Flow / Challenge-Request

### DISCUSSION THREAD

**Discussion by fue-medatixx** [Digitaler-Impfnachweis/certification-apis#52] [Q&A]
After the RU worked briefly yesterday at lunchtime, there has been a timeout since yesterday, late afternoon, for the challenge request against https://id.ru.impfproof.info
If you would just have a look ...

**Comment by arnoldmaik** [comment]
We unfortunately have a very similar problem. After everything worked until Thursday last week, we now always run into a timeout when we want to query a challenge. Are there any changes that we have missed or is this the status quo? Enclosed the request:

https://id.ru.impfnachweis.info/auth/realms/ubirch-certify/protocol/openid-connect/auth?redirect_uri=connector://authenticated&response_type=code&scope=openid&client_id=user-access-ti&nonce=1236547899521254656446

It is also possible that we are just making a silly mistake. In this case we would also be grateful for feedback.

**Comment by oliver-steinbrecher** [comment]
What kind of connector do you use?

**Reply by oliver-steinbrecher** [reply]
and Zugangsdienst

**Reply by fue-medatixx** [reply]
Secunet connector

**Reply by fue-medatixx** [reply]
curl "https://id.ru.impfnachweis.info/auth/realms/ubirch-certify/protocol/openid-connect/auth?redirect_uri=connector://authenticated&response_type=code&scope=openid&client_id=user-access-ti&nonce=123456789" -v
*   Trying 10.30.17.10...
* TCP_NODELAY set
* Connected to id.ru.impfnachweis.info (10.30.17.10) port 443 (#0)
* schannel: SSL/TLS connection with id.ru.impfnachweis.info port 443 (step 1/3)
* schannel: checking server certificate revocation
* schannel: sending initial handshake data: sending 188 bytes...
* schannel: sent initial handshake data: sent 188 bytes
* schannel: SSL/TLS connection with id.ru.impfnachweis.info port 443 (step 2/3)
* schannel: failed to receive handshake, need more data
* schannel: SSL/TLS connection with id.ru.impfnachweis.info port 443 (step 2/3)
* schannel: failed to receive handshake, SSL/TLS connection failed
* Closing connection 0
* schannel: shutting down SSL/TLS connection with id.ru.impfnachweis.info port 443
* Send failure: Connection was reset
* schannel: failed to send close msg: Failed sending data to the peer (bytes written: -1)
* schannel: clear security context handle
curl: (35) schannel: failed to receive handshake, SSL/TLS connection failed

**Reply by arnoldmaik** [reply]
Secunet & Arvarto

**Comment by oliver-steinbrecher** [comment]
We've succcessullfy tested it with a secunet connector

**Reply by arnoldmaik** [reply]
Maybe a routing problem:

route -p add 10.30.17.10 mask 255.255.255.0 [connector-ip] 

should work, right?

**Reply by oliver-steinbrecher** [reply]
yes that might be the issue - we changed to a different gematik network

**Comment by oliver-steinbrecher** [comment]
*   Trying 10.30.17.10:443...
* TCP_NODELAY set
* Connected to id.ru.impfnachweis.info (10.30.17.10) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: jurisdictionC=DE; jurisdictionST=Nordrhein-Westfalen; jurisdictionL=Koeln; businessCategory=Private Organization; serialNumber=HRB 82994; C=DE; ST=Nordrhein-Westfalen; L=Koeln; postalCode=50670; street=Im Mediapark 5; O=ubirch GmbH; CN=api.ru.impfnachweis.info
*  start date: May 10 14:15:13 2021 GMT
*  expire date: May 13 14:15:13 2022 GMT
*  subjectAltName: host "id.ru.impfnachweis.info" matched cert's "id.ru.impfnachweis.info"
*  issuer: C=DE; O=D-Trust GmbH; CN=D-TRUST SSL Class 3 CA 1 EV 2009
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x556564420e10)
> GET /auth/realms/ubirch-certify/protocol/openid-connect/auth?redirect_uri=connector://authenticated&response_type=code&scope=openid&client_id=user-access-ti&nonce=123456789 HTTP/2
> Host: id.ru.impfnachweis.info
> user-agent: curl/7.68.0
> accept: */*
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* Connection state changed (MAX_CONCURRENT_STREAMS == 250)!
< HTTP/2 200 
< cache-control: no-store, must-revalidate, max-age=0
< content-type: text/plain;charset=utf-8
< date: Wed, 02 Jun 2021 15:51:05 GMT
< location: https://id.certify.demo.ubirch.com/auth/realms/ubirch-certify/login-actions/authenticate?session_code=pupdficridX45cdZ09J2MKO1NBlrb6KJQL-4TM6BoEs&execution=fac97347-d739-43a5-adeb-1f600009e0ba&client_id=user-access-ti&tab_id=1VYNyCz3n48&auth_session_id=7de1a445-aed0-4083-a58d-cb044ee7ef83
< referrer-policy: no-referrer
< server: istio-envoy
< set-cookie: AUTH_SESSION_ID=7de1a445-aed0-4083-a58d-cb044ee7ef83.keycloak-certificato-1; Version=1; Path=/auth/realms/ubirch-certify/; SameSite=None; Secure; HttpOnly
< set-cookie: AUTH_SESSION_ID_LEGACY=7de1a445-aed0-4083-a58d-cb044ee7ef83.keycloak-certificato-1; Version=1; Path=/auth/realms/ubirch-certify/; HttpOnly
< set-cookie: KC_RESTART=eyJhbGciOiJIUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICIwZGNmM2ZmMy1mZjI4LTQ1YTYtOWY0ZC02NzBhMjRkYmRiNDkifQ.eyJjaWQiOiJ1c2VyLWFjY2Vzcy10aSIsInB0eSI6Im9wZW5pZC1jb25uZWN0IiwicnVyaSI6ImNvbm5lY3RvcjovL2F1dGhlbnRpY2F0ZWQiLCJhY3QiOiJBVVRIRU5USUNBVEUiLCJub3RlcyI6eyJzY29wZSI6Im9wZW5pZCIsImlzcyI6Imh0dHBzOi8vaWQuY2VydGlmeS5kZW1vLnViaXJjaC5jb20vYXV0aC9yZWFsbXMvdWJpcmNoLWNlcnRpZnkiLCJyZXNwb25zZV90eXBlIjoiY29kZSIsInJlZGlyZWN0X3VyaSI6ImNvbm5lY3RvcjovL2F1dGhlbnRpY2F0ZWQiLCJub25jZSI6IjEyMzQ1Njc4OSJ9fQ.BNXvmfGrYdWq0BjcmYtY0HDII8L5zdvfmAZw5MgXPbo; Version=1; Path=/auth/realms/ubirch-certify/; HttpOnly
< strict-transport-security: max-age=15552000; includeSubDomains; preload
< x-auth-challenge: 4a268722-191b-400f-b7ae-7a448c91ba7a
< x-content-type-options: nosniff
< x-envoy-decorator-operation: ruproxy.ubirch-prod.svc.cluster.local:8080/*
< x-envoy-upstream-service-time: 135
< x-xss-protection: 1; mode=block
< content-length: 0
< 
* Connection #0 to host id.ru.impfnachweis.info left intact
